### PR TITLE
Adapting to contracts v3.0.0-rc5

### PR DESCRIPTION
**Pull request by aaitor**
- documentation and methods to load NFT contracts
- Using the api to get the contract facade
- using latest versions on the workflow
- Renaming `transferForDelegate` to `claim`. It keeps the same functionality and footprint, just renames the method. `transferForDelegate` is there but marked as `@deprecated`
- Simplifying `claim` interface for different NFT implementations
- getting NFT balances using account or address
- draft: creation of contracts
- adapting nft interactions
- fixing merge conflict
- tests

## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: .github/workflows/testing-nightly.yml [hunk 1]
```diff
@@ -78,15 +62,33 @@ jobs:
...
+          name: nevermined-tools-output
+          path: tools/nevermined_tools.txt
+=======
+          IPFS_GATEWAY: https://ipfs.infura.io:5001
+          IPFS_PROJECT_ID: ${{ secrets.IPFS_PROJECT_ID }}
+          IPFS_PROJECT_SECRET: ${{ secrets.IPFS_PROJECT_SECRET }}
+          LOG_LEVEL: 1 # If LOG_LEVEL is >1 it will print the logger.debug calls
+          NO_GRAPH: true
+        run: yarn run integration:cover
+>>>>>>> feature/create_contracts
```
Comment (r-marques) [inline .github/workflows/testing-nightly.yml:82]: this is probably from an unsolved merge

File: .github/workflows/testing-nightly.yml [hunk 2]
```diff
@@ -7,61 +7,45 @@ on:
...
+        env:
+          ETH_PORT: 18545
+        run: |
+          yarn run test:cover
+      
       - name: Run integration tests
         env:
           SEED_WORDS: ${{ secrets.seedWords }}
           INFURA_TOKEN: ${{ secrets.INFURA_TOKEN }}
+<<<<<<< HEAD
```
Comment (r-marques) [inline .github/workflows/testing-nightly.yml:48]: unsolved merge

File: integration/nevermined/NFT721Templates.test.ts [hunk 3]
```diff
@@ -190,42 +193,38 @@ describe('NFT721Templates E2E', () => {
...
+                    serviceTypes: ['nft-sales', 'nft-access'],
+                    ercType: 721,
+                    nftType: NeverminedNFT721Type.nft721,
+                    nftContractAddress: nft.address,
+                    preMint: true,
+                    nftTransfer: false,
+                    royaltyAttributes: getRoyaltyAttributes(
+                        nevermined,
+                        RoyaltyKind.Standard,
+                        0
```
Comment (dancespiele) [inline integration/nevermined/NFT721Templates.test.ts:223]: supposedly the test says that wants to get 10% of the royalties but here is defined as 0

File: test/keeper/TestContractHandler.ts [hunk 4]
```diff
@@ -110,26 +98,45 @@ export default abstract class TestContractHandler extends ContractHandler {
...
+        let transactionResponse: TransactionResponse = await didRegistry.setNFT1155(
+            erc1155.address
+        )
+        let contractReceipt: ContractReceipt = await transactionResponse.wait()
         if (contractReceipt.status !== 1) {
-            throw new Error('Error calling "grantOperatorRole" on "erc721"')
-        }
+            throw new Error('Error calling "setNFT1155" on "didRegistry"')
+        }        
+        // let transactionResponse: TransactionResponse = await erc1155.grantOperatorRole(
```
Comment (dancespiele) [inline test/keeper/TestContractHandler.ts:127]: better remove commented code

File: test/keeper/conditions/TransferNFTCondition.test.ts [hunk 5]
```diff
@@ -128,11 +128,12 @@ describe('TransferNFTCondition', () => {
                 nftContractAddress: nftUpgradeable.address,
                 cap: BigNumber.from(100),
                 amount: nftAmount,
-                preMint: true
+                preMint: false
```
Comment (dancespiele) [inline test/keeper/conditions/TransferNFTCondition.test.ts:131]: don't we wanted to set `preMint` `true` by default?

File: test/keeper/conditions/TransferNFTCondition.test.ts [hunk 6]
```diff
@@ -340,7 +343,8 @@ describe('TransferNFTCondition', () => {
                 '',
                 activityId
             )
-            await didRegistry.mint(did, nftAmount, owner.getId())
+            await nevermined.nfts1155.mint(did, nftAmount, owner.getId(), owner)
+            // await didRegistry.mint(did, nftAmount, owner.getId())
```
Comment (dancespiele) [inline test/keeper/conditions/TransferNFTCondition.test.ts:347]: better to remove commented code

File: test/keeper/conditions/TransferNFTCondition.test.ts (hunk 5 above)
Comment (aaitor) [reply test/keeper/conditions/TransferNFTCondition.test.ts:131]: true or false is just a user option, here is false because we mint later

Comment (dancespiele) [reply test/keeper/conditions/TransferNFTCondition.test.ts:131]: I didn't see that it was just a test :)

**Review by dancespiele — APPROVED** [review]

**Pull request merged**